codec-http/src/main/java/io/netty/handler/codec/http/HttpMethod.java

```diff
@@ -115,12 +114,15 @@ public static HttpMethod valueOf(String name) {
             throw new IllegalArgumentException("empty name");
         }
 
+        return valueOfSantized(name);
+    }
+
+    /**
+     * Same as {@link #valueOf(String)} except the {@code name} parameter is known to be not {@code null} and trimmed.
+     */
+    public static HttpMethod valueOfSantized(String name) {
```

Comment:
@buchgr - Good question. Not sure that we do. It was done to preserve existing functionality...however it looks like all of this check is already duplicate logic of the HttpMethod constructor.